File: docs/snippets/all/archetypes/image_row_updates.py
"""
Update an image over time.

See also the `image_column_updates` example, which achieves the same thing in a single operation.
"""

import numpy as np
import rerun as rr

rr.init("rerun_example_image_row_updates", spawn=True)

for t in range(20):
    rr.set_time("time", sequence=t)

    image = np.zeros((200, 300, 3), dtype=np.uint8)
    image[:, :, 2] = 255
    image[50:150, (t * 10) : (t * 10 + 100)] = (0, 255, 255)

    rr.log("image", rr.Image(image))

File: docs/snippets/all/archetypes/image_row_updates.rs
//! Update an image over time.
//!
//! See also the `image_column_updates` example, which achieves the same thing in a single operation.

use ndarray::{Array, ShapeBuilder as _, s};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_image_row_updates").spawn()?;

    for t in 0..20 {
        rec.set_time_sequence("time", t);

        let mut image = Array::<u8, _>::zeros((200, 300, 3).f());
        image.slice_mut(s![.., .., 2]).fill(255);
        image
            .slice_mut(s![50..150, (t * 10)..(t * 10 + 100), 1])
            .fill(255);

        rec.log(
            "image",
            &rerun::Image::from_color_model_and_tensor(rerun::ColorModel::RGB, image)?,
        )?;
    }

    Ok(())
}
